### PR TITLE
Target net472

### PR DESCRIPTION
**Pull request by cdmihai**

### CODE REVIEW

**Comment by cdmihai** [comment]
@crummel, look ok for sourcebuild?

**Comment by cdmihai** [comment]
@mmitche, are there any pools with >dev15.7 and net472? We'll need net472 starting with this PR.

**Comment by crummel** [comment]
Looks good for source-build, thanks for pinging me!

File: netci.groovy [hunk 1]
```diff
@@ -107,7 +107,7 @@ def CreateJob(script, runtime, osName, isPR, machineAffinityOverride = null, sho
                     }
 
                     // this agent has VS 15.7 on it, which is a min requirement for our repo now.
-                    machineAffinityOverride = 'Windows.10.Amd64.ClientRS3.DevEx.Open'
+                    machineAffinityOverride = 'windows.10.amd64.clientrs4.devex.15.8.open '
```
Comment (rainersigwald) [inline netci.groovy]: We could also just drop Jenkins now, unless anyone would like to complain about VSTS?

File: src/MSBuild/app.amd64.config [hunk 2]
```diff
@@ -46,18 +46,6 @@
           <codeBase version="15.1.0.0" href="..\Microsoft.Build.Conversion.Core.dll"/>
         </dependentAssembly>
 
-        <!-- Redirects for facade assemblies -->
```
Comment (rainersigwald) [inline src/MSBuild/app.amd64.config]: Do we really not need these any more? I think I expected them to remain (maybe with an updated version) with no `codeBase`.

File: src/Package/MSBuild.Engine.Corext/MsBuild.Engine.Corext.nuspec [hunk 3]
```diff
@@ -29,8 +29,6 @@
     <file src="$X86BinPath$/Microsoft.Build.Tasks.Core.dll" target="v15.0/bin" />
     <file src="$X86BinPath$/Microsoft.Build.Utilities.Core.dll" target="v15.0/bin" />
     <file src="$X86BinPath$/System.Collections.Immutable.dll" target="v15.0/bin" />
-    <file src="$X86BinPath$/System.IO.Compression.dll" target="v15.0/bin" />
-    <file src="$X86BinPath$/System.Runtime.InteropServices.RuntimeInformation.dll" target="v15.0/bin" />
```
Comment (rainersigwald) [inline src/Package/MSBuild.Engine.Corext/MsBuild.Engine.Corext.nuspec]: This has implications for internal consumers, right? Has the internal cloud builder image moved to 4.7.2?

**Comment by cdmihai** [comment]
@dotnet-bot test this please

File: netci.groovy (hunk 1 above)
Comment (cdmihai) [reply netci.groovy]: I'd be in favor of that. The rhel 7.2 job is the only one left that needs porting. Do you know what queue to use for rhel 7.2 machines?

File: src/Package/MSBuild.Engine.Corext/MsBuild.Engine.Corext.nuspec (hunk 3 above)
Comment (cdmihai) [reply src/Package/MSBuild.Engine.Corext/MsBuild.Engine.Corext.nuspec]: Last time we talked they seemed OK with it.

File: src/MSBuild/app.amd64.config (hunk 2 above)
Comment (cdmihai) [reply src/MSBuild/app.amd64.config]: Very good question. Those types moved in-box, so I figured that if VS also moves to 472, they'll also start using the in-box types, in which case the redirects are not needed anymore. Is this a valid assumption?

File: src/Directory.Build.props [hunk 4]
```diff
@@ -23,7 +23,8 @@
     <Platforms>AnyCPU;x64</Platforms>
     
     <!-- Defaults for target frameworks and architecture -->
-    <LibraryTargetFrameworks>$(FullFrameworkTFM);netstandard2.0</LibraryTargetFrameworks>
```
Comment (cdmihai) [inline src/Directory.Build.props]: @crummel is it OK if msbuild does not build full framework on !windows?

Comment (crummel) [reply src/Directory.Build.props]: No, I think that will cause problems - SDK builds on the full framework to support using VS Code with a source-built SDK (issues https://github.com/dotnet/source-build/issues/125 and https://github.com/dotnet/sdk/issues/2249) and has a dependency on MSBuild, so we'll need the full-framework here too.  Would something like an `[En|Dis]ableFullFrameworkBuild` property that defaults to what you need and we can override make sense here?

Comment (cdmihai) [reply src/Directory.Build.props]: OK, added it back :)

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for CoreCLR
Previous: https://ci2.dot.net/job/Microsoft_msbuild/job/master/job/innerloop_Windows_NT_CoreCLR_prtest/895/

File: src/Directory.Build.props (hunk 4 above)
Comment (crummel) [reply src/Directory.Build.props]: Thanks!

**Pull request closed without merging**